README.md

```diff
@@ -1,66 +1,150 @@
+
+* `--version`: Prints current CSI external-snapshotter version and quits.
+
+* All glog / klog arguments are supported, such as `-v <log level>` or `-alsologtostderr`.
+
+
+## Upgrade from v1alpha1 to v1beta1
+
+The change from v1alpha1 to v1beta1 snapshot APIs is not backward compatible.
+
+If you have already deployed v1alpha1 snapshot APIs and external-snapshotter sidecar controller and want to upgrade to v1beta1, you need to do the following:
+* Delete volume snapshots created using v1alpha1 snapshot CRDs and external-snapshotter sidecar controller.
+* Uninstall v1alpha1 snapshot CRDs, external-snapshotter sidecar controller, and CSI driver.
+* Install v1beta1 snapshot CRDs, snapshot controller, CSI external-snapshotter sidecar and CSI driver.
+
+If you want to keep the physical snapshot resources created using the v1alpha1 version, here are suggested steps.
+Note: There is a possibility for data loss if something goes wrong during this process. Verify steps 1 and 2 works for a sample snapshot before moving forward. There was a bug in the pre-provisioned snapshots logic that was fixed in release [v1.2.1](https://github.com/kubernetes-csi/external-snapshotter/blob/release-1.2/CHANGELOG-1.2.md#notable-changes-1). Make sure you are using v1.2.1 or later before upgrading to v1beta1 if you want to try the following steps.
+1. Change deletionPolicy from Delete to Retain in the volume snapshot content.
+2. Delete volume snapshots created using v1alpha1 snapshot CRDs and external-snapshotter sidecar controller. Verify that volume snapshot content and the physical snapshot resource remain.
+3. Make sure above steps work for a sample snapshot before moving forward. This is to make sure that deletion of a volume snapshot object will not trigger the deletion of the volume snapshot content object and its associated physical snapshot resource on the storage system when the deletionPolicy is Retain. It is very important to verify this step to make sure there is no data loss.
+4. Uninstall v1alpha1 snapshot CRDs, external-snapshotter sidecar controller, and CSI driver. The v1alpha1 volume snapshot content and its associated physical snapshot resource will not be in the Kubernetes database any more.
+5. Install v1beta1 snapshot CRDs, snapshot controller, CSI external-snapshotter sidecar, and CSI drvier.
+6. Create a v1beta1 volume snapshot content API object for the physical snapshot resource.
+  * Note: Do not set the UID field of VolumeSnapshotRef! If you copied a volume snapshot content yaml file from your v1alpha1 setup, make sure to DELETE the UID field of VolumeSnapshotRef. This field will be populated by the controller automatically after a v1beta1 volume snapshot API object is created and bound to the volume snapshot content.
```

Comment:
are there any other things like annotations that they should also remove?

Reply:
Will remove this section as I have not tested it myself.